tests/test_inspect_pkg.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (C) 2014 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+from __future__ import annotations
+
+import json
+from pathlib import Path
+
+import pytest
+from conda import __version__ as conda_version
+from conda.core.prefix_data import PrefixData
+from packaging.version import Version, parse
+
+from conda_build.inspect_pkg import which_package
+
+
+@pytest.mark.skipif(
+    parse(conda_version) < Version("23.5.0"),
+    reason="tmp_env fixture first available in conda 23.5.0",
+)
+def test_which_package(tmp_path: Path):
+    # create a dummy environment
+    (tmp_path / "conda-meta").mkdir()
+    (tmp_path / "conda-meta" / "history").touch()
+    # a dummy package with a unique file (softlink) and a shared file (shared)
```

Comment:
```suggestion
    # a dummy package with a unique file (softlink) and a shared file (shared)
    (tmp_path / "shared").touch()
    (tmp_path / "softlink").symlink_to(tmp_path / "shared")
```

Reply:
we don't actually need the files to exist for `which_package` to work

Reply:
If we want to properly test for the symlink cases, we do.
Yes, `conda_build.utils.samefile` has a fallback so it works even if the files don't exist.
But here, to cover the case that one package reports multiple paths for the same file, we need the symlink so that `Path.samefile` will be true for hard- as well as symlink.

Reply:
you're right, I've expanded the softlink test to be more comprehensive